src/transformers/models/detr/modeling_detr.py

```diff
@@ -767,10 +768,10 @@ def forward(
 
             hidden_states, cross_attn_weights = self.encoder_attn(
                 hidden_states=hidden_states,
-                position_embeddings=query_position_embeddings,
+                object_queries=query_position_embeddings,
                 key_value_states=encoder_hidden_states,
                 attention_mask=encoder_attention_mask,
-                key_value_position_embeddings=position_embeddings,
+                spatial_position_embeddings=object_queries,
```

Comment:
Hmm this is confusing, we'd like to make the distinction between object queries (which are added to the inputs of each attention layer of the decoder) and spatial position embeddings (which are added to the keys and values of each cross-attention layer of the decoder)

Reply:
yeah, it seems a little confusing, since I was assigning object_queries to a variable different than object_queries arg. I think I confused thing. It should be:

```py
self.encoder_attn(
                hidden_states=hidden_states,
                object_queries=object_queries,  
                key_value_states=encoder_hidden_states,
                attention_mask=encoder_attention_mask,
                spatial_position_embeddings=query_position_embeddings,
                output_attentions=output_attentions,
            )
```